src/OrchardCore.Modules/OrchardCore.ContentFields/Extensions/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace OrchardCore.ContentManagement.Metadata.Models;
+
+public static class ContentPartFieldDefinitionExtensions
+{
+    /// <summary>
+    /// Returns the value of the defined content field from the <paramref name="contentItem"/>.
+    /// </summary>
+    public static TField GetContentField<TField>(
+        this ContentPartFieldDefinition fieldDefinition,
+        ContentItem contentItem)
+        where TField : ContentField =>
+        contentItem.Content[fieldDefinition.PartDefinition.Name] is JObject jPart &&
+        jPart[fieldDefinition.Name] is JObject jField &&
+        jField.ToObject<TField>() is { } field
```

Comment:
does `is { }` mean `!= null`. If so then `!= null` is easier to comprehend.

Reply:
It's a combined operation for pattern matching introduced in C# 8. It does `!= null` check and assignment to the `field` variable. So it makes the code more compact and lowers the cognitive complexity (once you are used to it anyway). Back in C# 7 we'd use `jField.ToObject<TField>() is var field` for the same, but starting C# 8 this empty object pattern is the norm.

If I replaced this with `!= null` then it would look like this:
```csharp
        jField.ToObject<TField>() != null
            ? jField.ToObject<TField>()
            : null;
```
I think that's less efficient and visually noisier. Please comment if you want C# 8 style, C# 7 style, or the old fashioned null check and explicit assignment.

Reply:
> and lowers the cognitive complexity (once you are used to it anyway)

Like pain, after a while you don't realize it's there, so pain isn't real actually.
We had a meeting today and nobody could explain what it does. Everyone could explain `!= null` though.

Please keep the old style, and add more lines if necessary for readability or prevent two calls to ToObject.

Reply:
It's not like pain, more like learning a new technique. First it's weird and awkward, but it you give it a chance you start seeing the utility in practice and it grows on you. Change is good.
(also we use this pattern matching in Lombiq everywhere)

But it's you guys' repo, so I abide by your decision and have rewrote the method.

Reply:
May be you can use `jField is not null` it’s more readable

Reply:
Ok. I have already eliminated the null check here, but I applied your suggestion to the null check in the other method in this file.

However this is starting to get a bit confusing. Does the OC repo have a style guide? I couldn't find it. If not, do you have plans to make one?

Reply:
https://github.com/OrchardCMS/OrchardCore/blob/main/CONTRIBUTING.md

Reply:
That links to [ASP.Net Core's engineering guidelines](https://github.com/dotnet/aspnetcore/wiki/Engineering-guidelines#coding-guidelines). If you can edit that, it would be good to add something about null checking conventions based on this discussion. If not, an OC-specific guide would still be valuable to keep a record of such decisions. Just a thought.